packages/typescript-eslint/src/config-helper.ts

```diff
@@ -118,8 +139,11 @@ export function config(
     );
     if (undefinedExtensions.length) {
       const configName =
-        configWithExtends.name != null
-          ? `, named "${configWithExtends.name}",`
+        config.name != null
+          ? `, named "${
+              // eslint-disable-next-line @typescript-eslint/no-base-to-string,@typescript-eslint/restrict-template-expressions
+              config.name
```

Comment:
Even though `config.name` is `unknown`, it should still be fine to concatenate it